common/src/main/java/io/pravega/example/tensorflow/TFObjectDetector.java

```diff
@@ -0,0 +1,141 @@
+import java.nio.FloatBuffer;
+import java.util.List;
+
+/**
+ * ObjectDetector class to detect objects using pre-trained models with TensorFlow Java API.
+ */
+public class TFObjectDetector implements Serializable {
+    private final Logger log = LoggerFactory.getLogger(TFObjectDetector.class);
+
+//    public List<Recognition> recognitions = null;
+//    Graph graph;
+    Session session;
+    // Params used for image processing
+    int SIZE = 416;
+    float MEAN = 255f;
+    Output<Float> output = null;
+    private List<String> LABEL_DEF;
+    private static TFObjectDetector single_instance = null;
+
+
+    public static TFObjectDetector getInstance() {
+        if(single_instance == null) {
+            single_instance = new TFObjectDetector();
+        }
+
+        return single_instance;
+    }
+
+    public TFObjectDetector() {
+        System.out.println("@@@@@@@@@@@  new TF @@@@@@@@@@@  " );
+
+        InputStream graphFile = getClass().getResourceAsStream("/tiny-yolo-voc.pb");       // The model
+        InputStream labelFile = getClass().getResourceAsStream("/yolo-voc-labels.txt");    // labels for classes used to train model
+
+        byte[] GRAPH_DEF = IOUtil.readAllBytesOrExit(graphFile);
+        LABEL_DEF = IOUtil.readAllLinesOrExit(labelFile);
+        Graph graph = new Graph();
+        graph.importGraphDef(GRAPH_DEF);
+        session = new Session(graph);
+        GraphBuilder graphBuilder = new GraphBuilder(graph);
+
+        output =
+                graphBuilder.div( // Divide each pixels with the MEAN
+                        graphBuilder.resizeBilinear( // Resize using bilinear interpolation
+                                graphBuilder.expandDims( // Increase the output tensors dimension
+                                        graphBuilder.decodeJpeg(
+                                                graph.opBuilder("Placeholder", "image")
+                                                        .setAttr("dtype", DataType.STRING)
+                                                        .build().output(0), 3),
+                                        graphBuilder.constant("make_batch", 0)),
+                                graphBuilder.constant("size", new int[]{SIZE, SIZE})),
+                        graphBuilder.constant("scale", MEAN));
+    }
+
+    /**
+     * Detect objects on the given image
+     *
+     * @param image the location of the image
+     * @return output image with objects detected
+     */
+    public byte[] detect(byte[] image) {
+        long start = System.currentTimeMillis();
+        byte[] finalData = null;
+
+
+        List<Recognition> recognitions = YOLOClassifier.getInstance().classifyImage(executeYOLOGraph(image), LABEL_DEF);
+
+        finalData = ImageUtil.getInstance().labelImage(image, recognitions);
+
+        long end = System.currentTimeMillis();
+        System.out.println("@@@@@@@@@@@  TENSORFLOW  TIME TAKEN FOR DETECTION @@@@@@@@@@@  " + (end - start));
```

Comment:
There are several time durations printed that have similar labels which is confusing. Where possible, label the duration with the method name.